### PR TITLE
indexer-common: Avoid using createdAt field for pagination

### PR DESCRIPTION
**Pull request by fordN**
## Changes
- Update several `NetworkMonitor` functions to paginate by `id` instead of by `createdAt`. Functions updated: subgraphs(), subgraphDeployments() and disputableAllocations(). 

## Background
On L2 networks, like arbitrum-one, `createdAt` is not necessarily unique, so paginating based on `createdAt` can lead to missing items.

### CODE REVIEW

**Review by dwerner — APPROVED** [review]
lgtm! thanks

File: packages/indexer-common/src/indexer-management/monitor.ts [hunk 1]
```diff
@@ -336,7 +336,7 @@ export class NetworkMonitor {
     }
     let subgraphs: Subgraph[] = []
     const queryProgress = {
-      lastCreatedAt: 0,
+      lastId: 0,
```
Comment (aasseman) [inline packages/indexer-common/src/indexer-management/monitor.ts:339]: The network subgraph's schema is using type `ID` for ids, which is an alias for `string`.
(see https://semiotic-ai.slack.com/archives/C032EDF093K/p1713996760318369?thread_ts=1713983989.035979&cid=C032EDF093K)

```suggestion
      lastId: '',
```

File: packages/indexer-common/src/indexer-management/monitor.ts [hunk 2]
```diff
@@ -352,10 +352,10 @@ export class NetworkMonitor {
       try {
         const result = await this.networkSubgraph.checkedQuery(
           gql`
-            query subgraphs($first: Int!, $lastCreatedAt: Int!, $subgraphs: [String!]!) {
+            query subgraphs($first: Int!, $lastId: Int!, $subgraphs: [String!]!) {
```
Comment (aasseman) [inline packages/indexer-common/src/indexer-management/monitor.ts:355]:
```suggestion
            query subgraphs($first: Int!, $lastId: ID!, $subgraphs: [String!]!) {
```

File: packages/indexer-common/src/indexer-management/monitor.ts [hunk 3]
```diff
@@ -596,10 +596,10 @@ export class NetworkMonitor {
       try {
         const result = await this.networkSubgraph.checkedQuery(
           gql`
-            query subgraphDeployments($first: Int!, $lastCreatedAt: Int!) {
+            query subgraphDeployments($first: Int!, $lastId: Int!) {
```
Comment (aasseman) [inline packages/indexer-common/src/indexer-management/monitor.ts:599]:
```suggestion
            query subgraphDeployments($first: Int!, $lastId: ID!) {
```

File: packages/indexer-common/src/indexer-management/monitor.ts [hunk 4]
```diff
@@ -1141,15 +1140,15 @@ Please submit an issue at https://github.com/graphprotocol/block-oracle/issues/n
             ) {
```
Comment (aasseman) [inline packages/indexer-common/src/indexer-management/monitor.ts:1140]: This is missing the variable declaration for `$lastId`. 
`graph-node` may let it pass, but I think that The Guild is/was working on making graph-node stricter wrt. the GraphQL spec, which could cause issues in the future.

**Review by aasseman — CHANGES_REQUESTED** [review]
Just nitpicking on adhering to the subgraph's schema, though graph-node may currently allow these things.

Also, somewhat unrelated, I found a few places in the code where pagination isn't implemented at all. In particular that could cause issues if somehow an indexer managed to open more than 1000 allocations. We know that some indexers are already indexing more than 1000 subgraphs at a time, so it's not completely impossible.

**Comment by dwerner** [comment]
Removing my approval as I've made changes now.

**Review by aasseman — CHANGES_REQUESTED** [review]
One of the comments was not addressed. GitHub won't let me explicitly select the piece of code....

So here's the issue:
`packages/indexer-common/src/indexer-management/monitor.ts:1139` `$createdAt: Int!` should be `$lastId: ID!`

**Comment by dwerner** [comment]
> One of the comments was not addressed. GitHub won't let me explicitly select the piece of code....
> 
> So here's the issue: `packages/indexer-common/src/indexer-management/monitor.ts:1139` `$createdAt: Int!` should be `$lastId: ID!`

Thanks for catching this.

**Review by aasseman — APPROVED** [review]

**Review by dwerner — APPROVED** [review]
Looks good to me.

**Pull request merged**